py-polars/polars/dataframe/frame.py

```diff
@@ -11333,13 +11333,22 @@ def show(
         │ 2   ┆ 7   ┆ b   │
         └─────┴─────┴─────┘
         """
-        df = self.head(n)
+        if limit is None:
+            df = self
+            tbl_rows = self.height
```

Comment:
Can set this to `-1` to display all rows (instead of `self.height`) 👌

Reply:
I find `self.height` more explicit but sure :+1: